Fix completion

aschrab opened the pull request:
A couple of small changes to have branch names completed for the `:Twiggy` command.

aschrab committed:
Fix references to branch completion function

aschrab committed:
Return all matching branches for completion

Completion functions registered as `custom` (as opposed to `customlist`)
are expected to return all available possibilities, allowing vim to do
the matching itself. Returning only the first match will often complete
too much (especially if completion is requested without a prefix being
entered first).

sodapopcan commented:
Wow, shows how much I use this feature!  Thanks for catching this and, yes, completion is much nicer this way.

I know it's a contentious issue but I absolutely can't stand when plugins expose random functions like you did with the complete function--I do realize it's public as it is, but the completion was one of the first things I did while I was still learning viml and I never changed it.  I would actually much prefer you move the completion function to the plugin file, make it a script variable and expose the `get_branches()` function which could be a really useful public function.

What do you think?  If it were to stay as is, it should at the very least be renamed to `twiggy#complete_git_branches()`.  How it is is a bit redundant.

Thanks again for catching this!

aschrab committed:
Expose get_branches() as Twiggy#get_branches()

In preparation for moving branch completion function to plugin file.

aschrab committed:
Don't expose branch completion function

Keep function used for completing branch names local to this plugin,
since it isn't likely to be useful elsewhere.

aschrab commented:
I generally find it difficult to decide between exposing that type of function which can be done in an autoload file, or keeping it private but needing to have that done during vim startup. So, my original choice here was to avoid moving it from where it was originally declared.

But since it is quite small I think I agree that it would be better to keep it private, so I've made the necessary adjustments to do so.

sodapopcan commented:
Thanks!
